roofit/roofitcore/src/RooAbsPdf.cxx

```diff
@@ -1027,6 +1027,12 @@ RooAbsReal* RooAbsPdf::createNLL(RooAbsData& data, const RooLinkedList& cmdList)
   pc.defineMutex("GlobalObservables","GlobalObservablesTag") ;
   pc.defineInt("NewStyle", "NewStyle", 0, 0);
 
+  if(pc.getSet("cPars") && pc.getSet("extCons")) {
+    std::string errMsg = "Specifying Constrain() when GlobalObservables() is passed is dangerous!";
+    coutE(InputArguments) << errMsg << std::endl;
+    throw std::invalid_argument(errMsg);
+  }
```

Comment:
Wouldn't it be easier to just use `pc.defineMutex` here rather than doing this manually?

Reply:
I don't actually want to get the exception, just print an error message. This change in RooAbsPdf is just to see in the CI how often `Constrain` is used together with `GlobalObservables`, the commit will be removed once I see there are none

Reply:
Ah, good idea!